### PR TITLE
Fix LinkedIn OAuth Scope and Profile Collection in GoTrue Library

### PR DESCRIPTION
**Pull request by tobias-istvan**
## What kind of change does this PR introduce?

This PR introduces a bug fix to address issues related to LinkedIn OAuth within the GoTrue library.

## What is the current behavior?

LinkedIn applications created after 1st of August experience difficulties while attempting to log in with GoTrue due to incorrect scope requests.

Relevant issue: [Link to relevant issue](https://github.com/supabase/gotrue/issues/1216#issuecomment-1688943690)

## What is the new behavior?

This PR aims to rectify the issue by making necessary adjustments to the OAuth scopes. Specifically, the scopes `openid`, `email`, and `profile` will be utilized. Additionally, the method of collecting profile information is updated, employing the `/v2/userinfo` API endpoint.

Visual changes: No visual changes.

## Additional context

While I may not be a Go developer or intimately familiar with the library, I have conducted local testing within a Docker environment. By utilizing the GoTrue.js library, I was able to successfully log in and retrieve the access token through the callback URL. However, I have not progressed to subsequent testing stages due to a lack of understanding about the next steps required.

### CODE REVIEW

File: internal/api/provider/linkedin.go [hunk 1]
```diff
@@ -1,156 +1,130 @@
...
 import (
-	"context"
-	"errors"
-	"strings"
+  "context"
+  "strings"
 
-	"github.com/supabase/gotrue/internal/conf"
-	"golang.org/x/oauth2"
+  "log"
```
Comment (kangmingtay) [inline internal/api/provider/linkedin.go:7]: this is unused and should be removed

**Review by kangmingtay — CHANGES_REQUESTED** [review]
i don't think we can change the existing linkedin oauth implementation because this will break existing linkedin oauth apps that used to work with the previous scopes

this API change only affects new linkedin oauth apps, if it needs to be supported, it should be created as a separated provider

**Comment by josmo** [comment]
@tobias-istvan I hope you don't mind but I opened https://github.com/supabase/gotrue/pull/1248 to create a new provider and keep the existing based on you're updates

@kangmingtay does that seem like what you had in mind? any feedback on it?

**Comment by kangmingtay** [comment]
Hey @tobias-istvan, thanks for spending the time and effort to contribute to this project! We've since merged the PR opened by @josmo #1248 since it preserves backward compatibility for existing linkedin oauth apps and will be closing this PR.

**Pull request closed without merging**